server/middleware/course-instances.coffee

```diff
@@ -198,13 +198,22 @@ module.exports =
       level = level.toObject({req: req})
     
     assessment = {}
-    if nextAssessmentOriginal
+    if nextAssessmentOriginal and not level.practice # Only show challenge after finishing however much practice we needed
       # Fetch full Assessment Level object
       dbq = Level.findOne({original: mongoose.Types.ObjectId(nextAssessmentOriginal)})
       dbq.sort({ 'version.major': -1, 'version.minor': -1 })
       dbq.select(parse.getProjectFromReq(req))
-      assessment = yield dbq
-      assessment = assessment.toObject({req: req})
+      # In parallel, fetch the level session for the Assessment Level to check if it's already complete
+      dbq2 = LevelSession.findOne({
+        'level.original': nextAssessmentOriginal + '',
+        creator: req.user.id + '',
+      })
```

Comment:
`dbq2` should be a more descriptive name like `sessionDbq`, maybe edit `dbq` to be `levelDbq` to match. Also, project this query so we just get the complete state.